### DISCUSSION TITLE
Marlin like M600

### DISCUSSION THREAD

**Discussion by epthewizard** [bassamanator/Sovol-SV06-firmware#23] [Ideas]
Hi, I just want to say I appreciate all the work you've put into this. The only thing I believe that is missing is a Marlin like color change. I cannot get the current one to work and I have tried many things. With Marlin I was easily able to throw in a M600 and watch as the printer paused, moved the extruder out of the way, unloaded the filament, reloaded the filament, reheated the hotend and then purged 50 mm or so before resuming the print. 

Like I said I've tried to recreate something like this but keep failing.

**Comment by bassamanator** [comment]
`M600` is used in the configuration, I've put it to use myself many times.

Expected behaviour:
1. Print pauses.
2. Toolhead moves up (if there's room to move up) and parks itself at the front.
3. Printer beeps 3 times.
4. Hotend temperature is set to 0C for safety.

You can then reheat the hotend, load filament either by hand and use `Extrude` section in the dashboard, once filament is loaded, click resume.

The only automation that I see as implementable, is the automatic unloading of filament, nothing more. Those other commands can be thrown in there, however, it would require a user that is in perfect sync with the machine, and I just don't see that happening.

I'm not quite sure what you've been failing at. Are you trying to write a macro to handle all the aforementioned steps?

**Reply by epthewizard** [reply]
Appreciate it. I was new to Klipper at the time and was expecting a filament change similar to how Marlin will unload, beep, cool down and wait for the new filament to be loaded, then heat up again once loaded and purge. I was able to figure it out though thanks

**Reply by bassamanator** [reply]
Welcome. If you have a solid solution for this, feel free to create a PR.
